src/mul/kernel/generic.jl

```diff
@@ -30,6 +34,22 @@ end
     return
 end
 
+@inline function subspace_mul_kernel_generic!(S::AbstractVector{T}, y, indices, U::Diagonal{N, Vector{N}}, k::Int, offset::Int) where {T, N}
+    @inbounds for i in axes(U, 1)
+        idx_i = k + indices[i] + offset
+        S[idx_i] = U[i, i] * S[idx_i]
+    end
+    return
+end
+
+@inline function subspace_mul_kernel_generic!(S::AbstractMatrix{T}, y, indices, U::Diagonal{N, Vector{N}}, k::Int, b::Int, offset::Int) where {T, N}
```

Comment:
why do we have a typevar `N` here? it's not used

Reply:
Yup, the typevar `N` is redundant. I have removed them. Thanks